### ISSUE TITLE
[DRG] Empathy copies buffs incorrectly

### ISSUE THREAD

**Issue by MB213** [EdenServer/community#3235]
### Checklist

- [x] I have searched for existing issues for issues like this one. The issue has not been posted. (Duplicate reports slow down development.)
- [x] I have provided reproducable steps. (No "as the title says" posts please. Provide reproducable steps even if it seems like a waste of time.)
- [x] I have provided my client version in the details. (type /ver into your game window)

### Details

/ver 30181205_0

The Dragoon merit ability Empathy appears to pick buffs to transfer randomly and existing buffs of the same type are overwritten. From 2007 sources onward, Empathy is said to copy buffs in order from left to right on the player screen, skipping over those which are already on the Wyvern: https://www.bg-wiki.com/index.php?title=Empathy&oldid=53140 (this behaviour predates the November 2007 patch but I referred to this date because it's more clearly explained).

Steps to reproduce as DRG75/WHM37:
1. With 3 points in Empathy, use Call Wyvern
2. Cast Protect, Shell, Stoneskin
3. Use Spirit Link
4. Wait for Spirit Link to cool down
5. Cast Barfira, Blink, Regen
6. Use Spirit Link
7. Observe that usually some of the first buffs will fall off (they are overwritten instead of being skipped over) and the corresponding number of new buffs are not transferred (you can tell which buffs the Wyvern got as they expire in the log).

There also seems to be an issue with two buffs of the same type not stacking when they should be able to. For example I have copied over Advancing March and Victory March and one immediately falls off. This doesn't seem to happen when songs are different types.

**Comment by MB213** [comment]
Reading more and there's a lot of conflicting information on this and it needs to be checked on retail. I'll leave it open for now in case someone can test it, but hopefully I have time to level DRG on retail at some point & I'll update/close if necessary.

**Comment by Spintimalli** [comment]
Following a meme dream of a super buffed Wyvern so i'm gonna get 5/5 Empathy and do some testing. (Un)fortunately I have never played post 2008 retail so I can't test there

**Comment by MB213** [comment]
Higana did some testing on retail and it seems as though it's random which buffs get copied.
BG has been completely wrong for like 13 years lol.  I don't know where they got that information. I was also not a Dragoon on retail, so this bug report is based on that false info.

The only difference we noted on retail is that songs will stack where as here they won't. That part is a bug. A wyvern can only have one copy of march here for example - if two transfer, one falls off.

**Comment by Spintimalli** [comment]
One thing I noticed from the list of buffs that can be transferred is COR rolls was not on it. I haven't tested to see or not, but wouldn't they be able to transfer to wyvern as well? Seems like an oversight on the wiki

**Comment by Satirof** [comment]
COR rolls can be copied, when I tested it as Mihki says; COR rolls were transferring. In fact I found that chaos rolls was transfering with 100% success every time. Other things seemed to be a bit more random. I feel there is probably a weighting system where certain buffs are given much higher priority, or 100% certainty first.

Whereas things like haste, haste 2, flurry, songs, prot, shell, bars, aqua, SS w.e seemed more random which would get transferred.

To get a full understanding of it would require multiple people spending hours testing everything (one person as drg, another as brd/cor, another as sch/rdm etc). I was using Trusts and was at the tail end of the last free login campaign, so my testing was limited both by jobs/npcs and time.
